client/lib/graph/query.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { isString, uniqueId } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { clearRequests } from 'state/requests/actions';
+
+const query = ( mapPropsToQuery ) => ( WrappedComponent ) => {
+	return class GraphQueryComponent extends Component {
+		state = {
+			results: null
+		};
+
+		uid = uniqueId();
+
+		static contextTypes = {
+			graph: PropTypes.object
+		};
+
+		constructor( props, context ) {
+			super( props, context );
+			this.buildQuery( props );
+		}
+
+		componentDidMount() {
+			this.unsubscribe = this.context.graph.store.subscribe( () => {
+				this.request();
```

Comment:
Can you explain why we trigger `this.request()` on every dispatch to the Redux store?